07b.2_school_build_region.R

```diff
@@ -9,20 +9,92 @@ if(!dir.exists(file.path(path_outputs_regional_notR, purpose, geography, region)
 
 start_time <- Sys.time() # for timing the script
 
-
 ###########################
-### SUBSET ZONES, CENTS AND LINES TO THE REGION
+### SUBSET ZONES & SCHOOLS TO THE REGION
 ###########################
 
-# Within-region zone
+# Within-region zone + school
 z <- z_all[z_all@data$lad11cd %in% region_lad_lookup$lad11cd, ]
+d <- d_all[d_all@data$lad11cd %in% region_lad_lookup$lad11cd, ]
+
+###########################
+### RUN RNET (UNLESS FLAGGED NOT TO IN BUILD PARAM CSV)
+###########################
+if (region_build_param$to_rebuild_rnet=="1") {
+  # DEFINE SCENARIOS
+  scenarios <- c("govtarget_slc","dutch_slc")
 
-# to do :add in subset of regions and schools, with shcools either via spatial look up or via LA code [latter safer?]
+  # IDENTIFY WITHIN-REGION FAST ROUTES
+  rnet_attributes <- rnet_all_attributes[(rnet_all_attributes$geo_code_o %in% z@data$geo_code) & (rnet_all_attributes$urn %in% d@data$urn), ]
+  summary({sel_rf <- rf_shape@data$id %in% rnet_attributes$id}) 
+  rf_rnet <- rf_shape[sel_rf,]  
+  rf_rnet@data <- data.frame(id = rf_rnet$id) 
+  rf_rnet@data <- left_join(rf_rnet@data, rnet_attributes, by="id")  
+  
+  # SUBSET BY MIN FLOW
+  rf_rnet <- rf_rnet[rf_rnet$all >= region_build_param$minflow_rnet, ]
+  rf_rnet <- rf_rnet[,c("id", "all", "bicycle", scenarios)]
+  
+  build_params[build_params$region_name == region, ]$n_flow_rnet <- nrow(rf_rnet)
+  build_params[build_params$region_name == region, ]$n_people_rnet <- sum(rf_rnet$all)
+  
+  # REMOVE ZERO FLOWS & SIMPLIFY LINE GEOMETRIES TO SPEED UP BUILD (needs mapshaper installed & available: https://github.com/mbloch/mapshaper/wiki/)
+  rf_rnet <- rf_rnet[(rf_rnet$dutch_slc > 0), ]
+  rf_rnet <- ms_simplify(input = rf_rnet, keep = region_build_param$rnet_keep, method = "dp", keep_shapes = TRUE, snap = TRUE)
+  
+  # BUILD RNET FOR BASELINE 
+  rnet <- overline(rf_rnet, "bicycle")
+  
+  # CREATE SCENARIO RESULTS IN A LIST IN PARALLEL
+  n_cores <- 4 # set max number of cores to 4
+  if(parallel:::detectCores() < 4) {n_cores <- parallel:::detectCores()}
```

Comment:
Good use of parallelisation. Sorry the super-fast version of overline was not ready in time for this.

Reply:
ha, the 'good use' is all yours – I just copied it from your old code :)